src/Microsoft.AspNet.Localization/RequestLocalizationMiddleware.cs

```diff
@@ -142,6 +154,39 @@ private CultureInfo GetCultureInfo(IList<string> cultures, IList<CultureInfo> su
                 }
             }
 
+            if (fallbackToAncestorCulture)
+            {
+                // Walk backwards through the culture list and remove any root cultures (those with no parent)
+                for (var i = cultureNames.Count - 1; i >= 0; i--)
+                {
+                    var cultureName = cultureNames[i];
+                    if (cultureName != null)
+                    {
+                        var lastIndexOfHyphen = cultureName.LastIndexOf('-');
+                        if (lastIndexOfHyphen > 0)
+                        {
+                            // Trim the trailing section from the culture name, e.g. "fr-FR" becomes "fr"
+                            cultureNames[i] = cultureName.Substring(0, lastIndexOfHyphen);
+                        }
+                        else
+                        {
+                            // The culture had no sections left to trim so remove it from the list of candidates
+                            cultureNames.RemoveAt(i);
+                        }
+                    }
+                    else
+                    {
+                        // Culture name was null so just remove it
+                        cultureNames.RemoveAt(i);
+                    }
+                }
+
+                if (cultureNames.Count > 0)
+                {
+                    return GetCultureInfo(cultureNames, supportedCultures, fallbackToAncestorCulture);
```

Comment:
So this would recurse through ["fr-FR-CA"] => ["fr-FR"] => ["fr"] trying to get the culture info for each one each time? Makes sense.